src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -69,7 +76,15 @@ public Uri ClientEndpoint
 
         internal string Version { get; }
 
-        internal AccessKey AccessKey { get; private set; }
+        internal AccessKey AccessKey
+        {
+            get
+            {
+                _accessKey ??= new AadAccessKey(_serviceEndpoint, _tokenCredential, ServerEndpoint);
```

Comment:
this is not thread safe get, does it matter?

Reply:
fixed